cmd/sourced/compose/workdir/workdir.go

```diff
@@ -123,18 +123,110 @@ func SetActive(reposdir string) error {
 	return os.Symlink(workdir, dir)
 }
 
-// Active returns th absolute path to $HOME/.srcd/workdirs/__active__
+// Active returns the absolute path to $HOME/.srcd/workdirs/__active__
 func Active() (string, error) {
 	return path(activeDir)
 }
 
+// ActiveRepoDir return repositories directory for an active working directory
+func ActiveRepoDir() (string, error) {
+	wpath, err := workdirsPath()
+	if err != nil {
+		return "", err
+	}
+	active, err := Active()
+	if err != nil {
+		return "", err
+	}
+	active, err = filepath.EvalSymlinks(active)
+	if err != nil {
+		return "", err
+	}
+
+	return stripBase(wpath, active)
+}
+
+// List returns array of working directories
+func List() ([]string, error) {
+	wpath, err := workdirsPath()
+	if err != nil {
+		return nil, err
+	}
+
+	dirs := make(map[string]bool)
+	err = filepath.Walk(wpath, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.Mode().IsRegular() {
```

Comment:
This will consider as a valid workdirectory any directory having a regular file, what is not necessarily correct.
I think that it could consider a valid workdirectory a directory having e.g. a `.DS_Store` file.

Reply:
true. Valid workdir is a directory with 2 files. `.env`&`docker-compose.yml`
I thought it would be overkill to check for both and checking for one doesn't guarantee more than checking for any file.
But I didn't think of `.DS_Store` file that might be created automatically while user uses Finder.
I'll update the PR checking for both files, thanks!